genesis/options/surfaces.py

```diff
@@ -215,52 +216,77 @@ def update_texture(
                 self.double_sided = double_sided
 
     def requires_uv(self):
-        return (
-            isinstance(self.get_texture(), ImageTexture)
-            or isinstance(self.opacity_texture, ImageTexture)
-            or isinstance(self.roughness_texture, ImageTexture)
-            or isinstance(self.metallic_texture, ImageTexture)
-            or isinstance(self.normal_texture, ImageTexture)
-            or isinstance(self.emissive_texture, ImageTexture)
+        textures = (
+            self.get_texture(),
+            self.opacity_texture,
+            self.roughness_texture,
+            self.metallic_texture,
+            self.normal_texture,
+            self.emissive_texture,
         )
+        return any(texture is not None and texture.requires_uv() for texture in textures)
+
+    def get_rgba(self, batch=False):
+
+        def fetch_batch(texture):
+            if isinstance(texture, BatchTexture):
+                return texture.textures if batch else texture.textures[:1]
+            else:
+                return [texture]
+
+        color_textures = fetch_batch(self.get_texture() if self.emissive_texture is None else self.emissive_texture)
+        opacity_textures = fetch_batch(self.opacity_texture)
```

Comment:
Do not skip line after function signature if there is no docstring.

Do not define nested function `fetch_batch`. Just do a for-loop:
```python
all_textures = []
for texture in (
    self.get_texture() if self.emissive_texture is None else self.emissive_texture,
    self.opacity_texture,
):
    textures = texture.textures if isinstance(texture, BatchTexture) else [texture]
    all_textures.append(textures if batch else textures[:1])
color_textures, opacity_textures = all_textures
```